### PR TITLE
Add metrics to BiDAF

### PR DESCRIPTION
**Pull request by matt-gardner**

### CODE REVIEW

File: allennlp/nn/initializers.py [hunk 1]
```diff
@@ -104,25 +104,26 @@ def __call__(self, module: torch.nn.Module) -> None:
...
         """
+        logger.info("Initializing parameters")
         # Store which initialisers were applied to which parameters.
         not_explicitly_initialized_parameters = []
         for name, parameter in module.named_parameters():
             is_initialized = False
             for initializer_regex, initializer in self._initializers.items():
                 if re.search(initializer_regex, name):
                     initializer(parameter)
-                    logger.info("Initializing %s using %s "
```
Comment (matt-gardner) [inline allennlp/nn/initializers.py]: If you're mostly using the default initializer, anyway, I don't think this is actually that helpful, because it doesn't tell you what it was actually initialized with.  But the motivation was to reduce test output, so I didn't get a huge log to page through before actually seeing my error.  I can just change the logging for this in the tests.

File: allennlp/training/metrics.py [hunk 2]
```diff
@@ -112,6 +123,73 @@ def reset(self):
...
+        ----------
+        predictions : ``torch.Tensor``, required.
+            A tensor of predictions of shape (batch_size, ...).
+        gold_labels : ``torch.Tensor``, required.
+            A tensor of integer class label of shape (batch_size, ...). It must be the same
+            shape as the ``predictions`` tensor.
+        mask: ``torch.Tensor``, optional (default = None).
+            Currently must be ``None``.  We keep this parameter just to match the API.
+        """
+        if mask is not None:
```
Comment (matt-gardner) [inline allennlp/training/metrics.py]: Yeah, it just wasn't obvious to me what the mask would do.  But you're right, it's not hard.  I'll put it in.

File: allennlp/common/testing/test_case.py [hunk 3]
```diff
@@ -24,7 +24,8 @@ class AllenNlpTestCase(TestCase):  # pylint: disable=too-many-public-methods
 
     def setUp(self):
         logging.basicConfig(format='%(asctime)s - %(levelname)s - %(name)s - %(message)s',
-                            level=logging.DEBUG)
+                            level=logging.INFO)
```
Comment (matt-gardner) [inline allennlp/common/testing/test_case.py]: By default, pytest shows: (1) the error, including stack traces, (2) stderr, and (3) stdout.  This means the logging statements show up at the bottom of the terminal, and because of all of our parameter and other logging when constructing models, a test involving BiDAF could have hundreds of lines of logging output that I have to scroll through before seeing the error.  It's pretty annoying.
